### PR TITLE
refactor dashboard to use notification context

### PR DESCRIPTION
**Pull request by jhonny-m**

### CODE REVIEW

File: apps/gp2-frontend/src/NotificationMessages.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  const addNotification = useCallback(
+    (notification: Notification) => {
+      setNotifications((prev) => [...prev, notification]);
+    },
+    [setNotifications],
+  );
+  const removeNotification = useCallback(
+    (notification: Notification) => {
+      setNotifications((prev) =>
+        prev.filter((n) => n.message !== notification.message),
```
Comment (markcsims) [inline apps/gp2-frontend/src/NotificationMessages.tsx:15]: Should we check the page here too? For example the same message is displayed on the dashboard and outputs.

File: packages/gp2-components/src/templates/PageNotifications.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  page,
+  children,
+}) => {
+  const { notifications, removeNotification } = useNotificationContext();
+  const pageNotifications = notifications.filter(
+    (notification) => notification.page === page,
+  );
+  const displayNotification = pageNotifications[0];
+  useEffect(
+    () => () => displayNotification && removeNotification(displayNotification),
```
Comment (markcsims) [inline packages/gp2-components/src/templates/PageNotifications.tsx:24]: Should this be run as part of the clean up code?

File: apps/gp2-frontend/src/AuthenticatedApp.tsx [hunk 3]
```diff
@@ -35,22 +40,33 @@ const AuthenticatedApp: FC<Record<string, never>> = () => {
...
     user?.onboarded || false,
   );
+
+  const welcomeBackMessage = `Welcome back to the GP2 Hub${
+    user?.firstName ? `, ${user.firstName}` : ''
+  }!`;
+
+  useEffect(() => {
+    if (showWelcomeBackBanner) {
+      addNotification({
```
Comment (markcsims) [inline apps/gp2-frontend/src/AuthenticatedApp.tsx:55]: Is this a hook? If so, it shouldn't be called conditionally?

Comment (jhonny-m) [reply apps/gp2-frontend/src/AuthenticatedApp.tsx:55]: the hook is `useNotificationContext`, the context has notifications plus functions to add/remove them

File: packages/gp2-components/src/templates/PageNotifications.tsx (hunk 2 above)
Comment (jhonny-m) [reply packages/gp2-components/src/templates/PageNotifications.tsx:24]: when the components gets dismounted, it cleans up the notification associated to the page,
I thought this was a simple way to dismiss it when we change the route without having to add a listener to the route

File: apps/gp2-frontend/src/NotificationMessages.tsx (hunk 1 above)
Comment (jhonny-m) [reply apps/gp2-frontend/src/NotificationMessages.tsx:15]: that's a valid point, I can create a function to validate the notifications

**Review by markcsims — APPROVED** [review]

**Pull request merged**